docs: Add `Development` and `Releases` sections to the documentation (#11932)
pola-rs/polars: docs/releases/upgrade/0.19.md

>>> pl.Series(values=15)
...
TypeError: Series constructor called with unsupported type 'int' for the `values` parameter
```

### Updates to expression input parsing

Methods like `select` and `with_columns` accept one or more expressions.
But they also accept strings, integers, lists, and other inputs that we try to interpret as expressions.
We updated our internal logic to parse inputs more consistently.

#### Example

Before:

```shell
>>> pl.DataFrame({'a': [1, 2]}).with_columns(None)
shape: (2, 1)
┌─────┐
│ a   │
│ --- │
│ i64 │
╞═════╡
│ 1   │
│ 2   │
└─────┘
```

After:

```shell
>>> pl.DataFrame({'a': [1, 2]}).with_columns(None)
shape: (2, 2)
┌─────┬─────────┐
│ a   ┆ literal │
│ --- ┆ ---     │
│ i64 ┆ null    │
╞═════╪═════════╡
│ 1   ┆ null    │
│ 2   ┆ null    │
└─────┴─────────┘
```

### `shuffle` / `sample` now use an internal Polars seed

If you used the built-in Python `random.seed` function to control the randomness of Polars expressions, this will no longer work.
Instead, use the new `set_random_seed` function.

#### Example

Before:

```python
import random

random.seed(1)
```

After:

```python
import polars as pl

pl.set_random_seed(1)
```

## Deprecations

> Creating a consistent and intuitive API is hard, finding the right name for each function, method, and parameter might be the hardest part.

Reviewer: ```suggestion
Creating a consistent and intuitive API is hard; finding the right name for each function, method, and parameter might be the hardest part.
```

> The new version comes with quite some naming changes, and you will most likely run into deprecation warnings when upgrading to `0.19`.

Reviewer: ```suggestion
The new version comes with several naming changes, and you will most likely run into deprecation warnings when upgrading to `0.19`.
```

If you want to upgrade without worrying about deprecation warnings right now, you can add the following snippet to your code:

```python
import warnings

warnings.filterwarnings("ignore", category=DeprecationWarning)
```

### `groupby` renamed to `group_by`

> This is not a change we make lightly, as it will impact almost all our users. But "group by" are really two different words, and our naming strategy dictates that these should be separated by an underscore.

Reviewer: ```suggestion
This is not a change we make lightly, as it will impact almost all our users. But "group by" is really two different words, and our naming strategy dictates that these should be separated by an underscore.
```
